warehouse/integrations/deltalake/deltalake.go

```diff
@@ -526,6 +527,15 @@ func (d *Deltalake) tableLocationQuery(tableName string) string {
 
 // AddColumns adds columns to the table.
 func (d *Deltalake) AddColumns(ctx context.Context, tableName string, columnsInfo []warehouseutils.ColumnInfo) error {
+	tableSchema, err := d.fetchTableAttributes(ctx, tableName)
+	if err != nil {
+		return fmt.Errorf("fetch table attributes: %w", err)
+	}
+	columnsInfo = lo.Filter(columnsInfo, func(columnInfo warehouseutils.ColumnInfo, _ int) bool {
```

Comment:
1. minor: `columnsInfo` essentially are missing columns from tableschema fetched from warehouse? If yes, would `missingColumnsInfo` or `toAddColumnsInfo` be helpful var name?

2. How would idea of fetching table schema at higher cadence solution we are working on impact this change? Or is it in such cases of discrepancy between event schema and local table schema, we will always fetch table schema from warehouse?

Reply:
1. Renamed output of filtering to `columnsToAddInfo`
2. If there is a discrepancy between event schema and local table schema, we currently don't fetch the table schema from the warehouse. The diff is computed and an ALTER statement is executed for missing tables. So I would say schema v2 implementation is independent of this change